### PR TITLE
PRSDM-4416 - Scroll-spy not expanding menu on Apple site

### PR DESCRIPTION
**Pull request by WiaanB**
### Description
Ensured that only the defined targets get identified

### Issue
[PRSDM-4416](https://spandigital.atlassian.net/browse/PRSDM-4416)

### Screenshots

https://github.com/SPANDigital/presidium-js/assets/34862470/bdddfd8a-fdd5-4e6f-bef5-b57710261a40

### Checklist before merging

* [ ] Is this code covered by tests?
* [ ] Is the documentation updated for this change?

### CODE REVIEW

File: src/util/scroll-spy.js [hunk 1]
```diff
@@ -334,7 +334,7 @@
 
 				// Get the content for the nav item
                 const target = item.getAttribute(options.attribute)
-				const id = decodeURIComponent(target.substr(1));
+				const id = decodeURIComponent(target?.substr(1));
```
Comment (meyerhp) [inline src/util/scroll-spy.js:337]: Happy with the change. But would like to know why the target is null?

Comment (WiaanB) [reply src/util/scroll-spy.js:337]: The element does not have a "data-slug" attribute on it. (Target, and ID are both the "data-slug" attribute on the elements)
Screenshot 2023-09-27 at 08 07 10

---

From my experience working with Presidium, I haven't come across a file being referenced this way, and I'm not entirely sure why it does not have a "data-slug". This seems like an issue unique to this repo, thoughts?
Screenshot 2023-09-27 at 08 07 59

**Comment by WiaanB** [comment]
After further investigation, clicking on the Parent element of a deeply nested tree (such as "Releases" in this image below)
would not activate on the first child of content ("Core" in the example). To do so, selecting the first element would result in returning the correct behaviour.
image 

https://github.com/SPANDigital/presidium-js/assets/34862470/8fa933bd-37d9-45de-94cb-971a56d8dbe0

**Review by CharlRitterCompany — APPROVED** [review]

**Review by kmorake — APPROVED** [review]

**Review by christopherbrunsdon — APPROVED** [review]
Tested on local ... OK
👍

**Pull request merged**